### PR TITLE
SLCORE-794 Analysis in SLOOP - analysis triggering

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: API_CHANGES.md [hunk 1]
```diff
@@ -12,8 +28,12 @@
   * Analysis will be disabled for languages specified `disabledLanguagesForAnalysis` but it will be still possible to consume Rule Descriptions
   * Can be null or empty if clients do not wish to disable analysis for any loaded plugin
 
-
 ## New features
+* Add a method to `org.sonarsource.sonarlint.core.rpc.protocol.SonarLintRpcClient` to allow backend to request inferred analysis properties from the client before every analysis. It's important because properties may change depending on files being analysed.
+  * `org.sonarsource.sonarlint.core.rpc.protocol.SonarLintRpcClient#getInferredAnalysisProperties` to request inferred properties
+* Add a method to the `org.sonarsource.sonarlint.core.rpc.protocol.backend.analysis.AnalysisRpcService` to let the client notify the backend with user defined analysis properties
+  * `org.sonarsource.sonarlint.core.rpc.protocol.backend.analysis.AnalysisRpcService#didSetUserAnalysisProperties` to set user defined properties
+* For analysis, both user-defined and inferred properties will be merged. If the same property is inferred by the client and provided by the user - the inferred value will be used for analysis.
```
Comment (damien-urruty-sonarsource) [inline API_CHANGES.md:36]: I think this section and the next 2 ones are misplaced. They should belong to 10.4

File: API_CHANGES.md [hunk 2]
```diff
@@ -22,13 +42,26 @@
...
+### File events
+
+* Add the `didOpenFile` and `didCloseFile` methods to `org.sonarsource.sonarlint.core.rpc.protocol.backend.file.FileRpcService`.
+  * Clients are supposed to call these methods when a file is opened in the editor or closed.
+
+### Analysis
+
+* Add a new constructor in `org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.InitializeParams` to let clients provide if automatic analysis is enabled.
+* Add a new `didChangeAutomaticAnalysisSetting` method in `org.sonarsource.sonarlint.core.rpc.protocol.backend.analysis.AnalysisRpcService`
+  * Clients are expected to call it whenever users change the "enable automatic analysis" setting.
```
Comment (damien-urruty-sonarsource) [inline API_CHANGES.md:54]: We should also document the new `analyze*` methods in the analysis service

**Review by damien-urruty-sonarsource — APPROVED** [review]
Minor comments on the docs, else LGTM

File: API_CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,19 @@
+# 10.4
+
+## Breaking changes
+
+* Add new `isUserDefined` parameter into `org.sonarsource.sonarlint.core.rpc.protocol.common.ClientFileDto`
+  * User defined files will be included in analysis. Non-user-defined files such as generated or library files will be excluded from
+    analysis.
```
[the remark this answers is not in the document]
Comment (kirill-knize-sonarsource) [reply API_CHANGES.md:7]: Yes, thanks. Changed.

File: API_CHANGES.md (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply API_CHANGES.md:52]: Moved.

File: API_CHANGES.md (hunk 2 above)
Comment (kirill-knize-sonarsource) [reply API_CHANGES.md]: Added.

**Pull request merged**